bin/util.py

```diff
@@ -96,13 +106,19 @@ def error(msg):
 
 def fatal(msg, *, force=threading.active_count() > 1):
     print(f'\n{Fore.RED}FATAL ERROR: {msg}{Style.RESET_ALL}', file=sys.stderr)
-    if force:
-        sys.stderr.close()
-        sys.stdout.close()
-        # exit even more forcefully to ensure that daemon threads dont break something
-        os._exit(1)
-    else:
-        sys.exit(1)
+    exit(force)
+
+
+def message(msg, task=None, item=None, *, color=''):
+    if task is not None:
+        print(f'{Fore.CYAN}{task}: {Style.RESET_ALL}', end='', file=sys.stderr)
+    if item is not None:
+        print(item, end='   ', file=sys.stderr)
+    print(f'{color}{msg}{Style.RESET_ALL}', file=sys.stderr)
+    if color == Fore.YELLOW:
+        config.n_warn += 1
+    if color == Fore.RED:
+        config.n_error += 1
```

Comment:
adding this to the existing log,warn... functions is a bit weird since they print other stuff.
I could add separate functions but thats also weird... i am not super happy with any of these... 
We could also define our own "colors" and only if those are passed change the counter?

Reply:
That's already better than passing terminal color codes, thanks :slightly_smiling_face: